karyon2-admin-plugins/karyon2-admin-healthcheck-plugin/src/main/java/netflix/adminresources/HealthCheckResource.java

```diff
@@ -0,0 +1,39 @@
+package netflix.adminresources;
+
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import netflix.karyon.health.HealthCheckInvocationStrategy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.concurrent.TimeoutException;
+
+@Path("/healthcheck")
+@Produces(MediaType.TEXT_HTML)
+@Singleton
+public class HealthCheckResource {
+    private static final Logger logger = LoggerFactory.getLogger(HealthCheckResource.class);
+    private HealthCheckInvocationStrategy invocationStrategy;
+
+    @Inject
+    public HealthCheckResource(HealthCheckInvocationStrategy invocationStrategy) {
```

Comment:
I would actually remove the strategy and have the handler invoked directly. Strategy isn't in general useful & I have removed the async strategy from 2.x